### PR TITLE
refactor(ui): refactor ui i18n

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Refactor ui i18n logic to pick up the latest i18n design

- add new API `/getPhrase`
- remove the UI's build in resource loading logic, using the resource and language returned from the phrase api
- use build-in en as default fallback language
-  update the preview logic to load resource dynamically from the api

## Testing

test locally
@xiaoyijun @IceHe

### CODE REVIEW

File: packages/ui/src/i18n/utils.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+  );
+
+  return languageDetector.detect();
+};
+
+// Must be called after i18n's initialization
+export const changeLanguage = async (targetLanguage: string) => {
+  const { resources, lng } = await getI18nResource(targetLanguage);
+  console.log(resources, lng);
```
Comment (xiaoyijun) [inline packages/ui/src/i18n/utils.ts:56]: Shall we remove this line?

File: packages/ui/src/types/index.ts [hunk 2]
```diff
@@ -34,7 +33,7 @@ export enum ConfirmModalMessage {
 
 export type PreviewConfig = {
   signInExperience: SignInExperienceSettingsResponse;
-  language: LanguageKey;
+  language: string;
```
Comment (xiaoyijun) [inline packages/ui/src/types/index.ts:36]: Maybe `LangaugeTag`?

File: packages/ui/src/i18n/utils.ts (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/i18n/utils.ts:56]: yes

File: packages/ui/src/types/index.ts (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/types/index.ts:36]: nope technically it is loaded from a global post message response, can't guarantee it's type

File: packages/ui/src/i18n/utils.ts (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/i18n/utils.ts:56]: udpated

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Pull request merged**